src/utils.h

```diff
@@ -377,5 +377,160 @@ Rescale(Eigen::ArrayXXd               weights,
   intercept_archive.emplace_back(std::move(intercept));
 }
 
+
+//' Return a vector of random index
+//' 
+//' @param n_samples the sampler uses range from 0 to n_samples
+//' @param length the number of random index
+
+Eigen::ArrayXXi
+IndexStochastic(const unsigned n_samples,
+                const unsigned length)
+{
+  Eigen::ArrayXXi index(1, length);
+  for (unsigned i = 0; i < length; ++i){
+    unsigned s_ind = floor(R::runif(0.0, n_samples));
+    index.col(i) = s_ind;
+  }
+  return(index);
```

Comment:
```suggestion
  return index;
```

Try to use a consistent coding style. Most of the code currently uses `return foo;` rather than `return(foo);`, so please try to stick to this scheme.